src/include/units/bits/external/fixed_string.h

```diff
@@ -27,9 +27,159 @@
 
 namespace units {
 
-template<typename CharT, std::size_t N>
+template<typename CharT, std::size_t N, std::size_t M>
 struct basic_fixed_string {
   CharT data_[N + 1] = {};
+  char ascii_data_[M + 1] = {};
+
```

Comment:
I am not the biggest fan of `basic_fixed_string` storing 2 string inside. We should either rename it to something called `basic_symbol_text` or do a wrapper over it as I did here: https://godbolt.org/z/YQxKju. Personally, I like the second option more.

Reply:
It also makes `basic_fixed_string` being an implementation detail which is very good for standardization needs. We just need to standardize a "pair" of those ;-) (note that getters return auto which will mean an "implementation-defined" type)

Reply:
Oh, and of course a new header for this will be needed. Please put it directly under `units` folder as it is a part of library public interface that has to be standardized.

Reply:
> I am not the biggest fan of `basic_fixed_string` storing 2 string inside. We should either rename it to something called `basic_symbol_text` or do a wrapper over it as I did here: https://godbolt.org/z/YQxKju. Personally, I like the second option more.

I see. The partial template specialization I did was a bit messy. However, in your example, two basic_fixed_strings are stored regardless of whether the unit actually has two different symbols or not. Isn't there a more efficient way to do this? Or is the duplication unavoidable / optimized away in some way?

Reply:
It is compile-time only code (it does not exist in runtime). If you are worried about compile-time performance of copying characters to 2 buffers instead of one, alternatively for unified value you can:
- provide `0` as the second size and store the data only in the first one
- add `if constexpr(M == 0)` branch to `ascii()` getter to return `standard` text rather than the `ascii` one.

I am fine with mine or the above solution. Please pick whichever you like (will be easier to implement and maintain).

Reply:
> It is compile-time only code (it does not exist in runtime). If you are worried about compile-time performance of copying characters to 2 buffers instead of one, alternatively for the case of unified value (the same for both strings) you can:
> 
>     * provide `0` as the second size and store the data only in the first one
> 
>     * add `if constexpr(M == 0)` branch to `ascii()` getter to return `standard` text rather than the `ascii` one.
> 
> 
> I am fine with both (previously suggested and the above) solutions. Please pick whichever you like (will be easier to implement and maintain).

Oh, right. So they exist only for the purpose of the end result, e.g. formatting? In that case, it wouldn't matter much at all. I'm going for the more straightforward one then if the second way ends up being tedious.